src/components/deploymentConfig/DeploymentTemplateView.tsx

```diff
@@ -580,31 +591,55 @@ export const DeploymentTemplateEditorView = ({
     }, [selectedChart, environments])
 
     useEffect(() => {
-        if (charts.length > 0 && !globalChartRef) {
-            setGlobalChartRef(charts.find((_chart) => _chart.id === globalChartRefId))
+        if (selectedChart && charts.length > 0) {
+            const _filteredCharts = charts
+                .filter((chart) => {
+                    if (!globalChartRef && chart.id === globalChartRefId) {
+                        setGlobalChartRef(chart)
+                    }
+
+                    return chart.id !== selectedChart.id && chart.name === selectedChart.name
+                })
+                .sort((a, b) => versionComparator(a, b, 'version', SortingOrder.DESC))
+
+            setFilteredCharts(
+                _filteredCharts.map((chart) => ({
+                    id: `version-${chart.version}`,
+                    label: chart.version,
+                    value: chart.id,
+                    kind: 'chartVersion',
+                })) as DeploymentChartOptionType[],
+            )
         }
-    }, [charts])
+    }, [selectedChart, charts])
 
     useEffect(() => {
         if (selectedChart && selectedOption && selectedOption.id !== -1 && !fetchedValues[selectedOption.id]) {
             setFetchingValues(true)
             const isEnvOption = selectedOption.kind === 'env'
-            const _getDeploymentTemplate =
-                isEnvOverride || isEnvOption
-                    ? getEnvDeploymentTemplate(appId, isEnvOption ? selectedOption.id : envId, selectedOption.value)
-                    : getDeploymentTemplate(+appId, +selectedOption.value)
+            const isChartVersionOption = selectedOption.kind === 'chartVersion'
```

Comment:
Can we keep all kind in constant?

Reply:
Moved this & others to const that can be moved safely